src/Features/CSharp/Portable/CodeRefactorings/NodeSelectionHelpers.cs

```diff
@@ -4,22 +4,63 @@
 {
     internal static class NodeSelectionHelpers
     {
-        internal static async Task<SyntaxNode> GetSelectedDeclarationOrVariableAsync(CodeRefactoringContext context)
+        internal static async Task<ImmutableArray<SyntaxNode>> GetSelectedDeclarationsOrVariablesAsync(CodeRefactoringContext context)
         {
-            // Consider:
-            // MemberDeclaration: member that can be declared in type (those are the ones we can pull up) 
-            // VariableDeclaratorSyntax: for fields the MemberDeclaration can actually represent multiple declarations, e.g. `int a = 0, b = 1;`.
-            // ..Since the user might want to select & pull up only one of them (e.g. `int a = 0, [|b = 1|];` we also look for closest VariableDeclaratorSyntax.
-            return await context.TryGetRelevantNodeAsync<MemberDeclarationSyntax>().ConfigureAwait(false) as SyntaxNode ??
-                await context.TryGetRelevantNodeAsync<VariableDeclaratorSyntax>().ConfigureAwait(false);
+            var (doc, span, cancellationToken) = context;
+            if (span.IsEmpty)
+            {
+                // if the span is empty then we are only selecting one "member" (which could include a field which declared multiple actual members)
+                // Consider:
+                // MemberDeclaration: member that can be declared in type (those are the ones we can pull up) 
+                // VariableDeclaratorSyntax: for fields the MemberDeclaration can actually represent multiple declarations, e.g. `int a = 0, b = 1;`.
+                // ..Since the user might want to select & pull up only one of them (e.g. `int a = 0, [|b = 1|];` we also look for closest VariableDeclaratorSyntax.
+                var memberDeclaration = await context.TryGetRelevantNodeAsync<MemberDeclarationSyntax>().ConfigureAwait(false);
+                if (memberDeclaration == null)
+                {
+                    // could not find a member, we may be directly on a variable declaration
+                    var varDeclarator = await context.TryGetRelevantNodeAsync<VariableDeclaratorSyntax>().ConfigureAwait(false);
+                    if (varDeclarator == null)
+                    {
+                        return ImmutableArray<SyntaxNode>.Empty;
+                    }
+                    else
+                    {
+
+                        return ImmutableArray.Create(varDeclarator as SyntaxNode);
+                    }
+                }
+                else
+                {
+                    return memberDeclaration switch
+                    {
+                        FieldDeclarationSyntax fieldDeclaration => fieldDeclaration.Declaration.Variables.Cast<SyntaxNode>().AsImmutable(),
+                        EventFieldDeclarationSyntax eventFieldDeclaration => eventFieldDeclaration.Declaration.Variables.Cast<SyntaxNode>().AsImmutable(),
+                        _ => ImmutableArray.Create(memberDeclaration as SyntaxNode),
```

Comment:
```suggestion
                        EventFieldDeclarationSyntax eventFieldDeclaration => eventFieldDeclaration.Declaration.Variables.CastArray<SyntaxNode>(),
                        _ => ImmutableArray.Create<SyntaxNode>(memberDeclaration),
```

Note: we never use `as` in our codebase, unless it is for a case where the type-cast coudl fail and we then *will* check it for null.